File: app/src/main/java/com/extremedesign/typeracer/DataRepository/IDataBaseImplementations/IDataSource.java
package com.extremedesign.typeracer.DataRepository.IDataBaseImplementations;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;

import com.extremedesign.typeracer.listener.JobWorker;
import com.extremedesign.typeracer.listener.ProfileImageListener;
import com.extremedesign.typeracer.model.ProfileImage;
import com.extremedesign.typeracer.model.User;

import java.util.List;

public interface IDataSource {
//    void getData(String amount, String category, String difficulty, String type, final CustomListenerRep listenerRep);
//
//    void insertData(List<Question> questionList);
//
//    void getListOfCategories(CategoryListener listener);

    void insertUser(User user);

    LiveData<User> getCurrentUser(String uid);

    void insertDataProfileImage(ProfileImage profileImage, JobWorker listener);

    void insertDataProfileImages(List<ProfileImage> profileImages);

    void getDataProfileImages(ProfileImageListener listener);

    void getProfileImageByName(String name,ProfileImageListener listener);

}

File: app/src/main/java/com/extremedesign/typeracer/data_repository/i_database_implementations/IDataSource.kt
package com.extremedesign.typeracer.data_repository.i_database_implementations

import androidx.lifecycle.LiveData
import com.extremedesign.typeracer.listener.JobWorker
import com.extremedesign.typeracer.listener.ProfileImageListener
import com.extremedesign.typeracer.model.ProfileImage
import com.extremedesign.typeracer.model.User

interface IDataSource {
    fun insertUser(user: User?)
    fun getCurrentUser(uid: String?): LiveData<User?>?
    fun insertDataProfileImage(profileImage: ProfileImage?, listener: JobWorker?)
    fun insertDataProfileImages(profileImages: List<ProfileImage?>?)
    fun getDataProfileImages(listener: ProfileImageListener?)
    fun getProfileImageByName(name: String?, listener: ProfileImageListener?)
}